packages/evm/src/interpreter.ts

```diff
@@ -48,16 +49,26 @@ export interface Env {
   codeAddress: Address /* Different than address for DELEGATECALL and CALLCODE */
   gasRefund: bigint /* Current value (at begin of the frame) of the gas refund */
   containerCode?: Buffer /** Full container code for EOF1 contracts */
+  eofContainer?: EOFContainer // Only EOF contracts
+}
+
+type eip4750ReturnStackItem = {
+  codeSectionIndex: number
+  offset: number // PC post instruction
+  stackHeight: number
 }
 
 export interface RunState {
   programCounter: number
+  programCounterOffset: number // Used in EIP 4750. This is used to correctly report the `PC` opcode
   opCode: number
   memory: Memory
   memoryWordCount: bigint
   highestMemCost: bigint
   stack: Stack
-  returnStack: Stack
+  returnStack: Stack // Used in EIP 2315
+  returnStackEIP4750: eip4750ReturnStackItem[] // Used in EIP 4750 (these two EIPs could co-exist)
```

Comment:
I would suggest to use the generic name for 4750 and instead do `returnStack2315` since 2315 is the one (very much) less likely to get practical relevance (and we can do such changes since marked as `experimental`).